### PR TITLE
weather@mockturtl Update 3.4.2

### PR DESCRIPTION
**Pull request by Gr3q**
This is a big enough PR already (excluding translations...) to put new features in so they go to the next version.

* Fix https://github.com/linuxmint/cinnamon-spices-applets/issues/5419
* Fix https://github.com/linuxmint/cinnamon-spices-applets/issues/5680

Changes
* Made HTTP requests cancellable and enforce using them
* All HTTP requests are now have  
* Simplify HTTPLib call signatures (and removing intermediary calls from `main.ts`
* All data refresh goes through the main loop now to avoid parallel refreshing
* If the loop is requested to refresh immediately by the user it cancels the previous one if it's in progress and immediately begins a new one.
* Loop only gives 30 seconds for each refresh on JS level, then it cancels them to avoid the loop deadlocking.
* Fix all instances where it's possible not to resolve promisified Glib/Gio/Soup library async functions by adding try/catch inside.
* Use `https://nominatim.openstreetmap.org/` as a GeoCode backend instead of the default `https://nominatim.gnome.org/`

- [x] Tested on Cinnamon 3.8 but I will test it again when the PR is ready.

### CODE REVIEW

**Comment by Gr3q** [comment]
@altaloma @autumnlansing  

I will not take this out of draft until someone tests this PR because how much it changes. It doesn't have to be you, you guys reported the issues so it would be helpful. 

Things need testing:
* All setting changes are properly reflected in the UI immediately 
* Errors are displayed the same as before
* Applet should not end up in a state where it stops the refreshing unless it's an user error (bad key, not key, location not covered etc)

This should be extracted where the applet is installed (`~/.config/cinnamon/spices/` if you are on a newer version of Cinnamon, I don't remember the old location), then Cinnamon must be restarted. Debug logs are useful.
[weather@mockturtl-3.4.2.zip](https://github.com/linuxmint/cinnamon-spices-applets/files/15179337/weather%40mockturtl.zip)

Wish there was a way for easier way to share release candidates with people who want to test newer version of the applets.

File: weather@mockturtl/files/weather@mockturtl/po/ca.po [hunk 1]
```diff
@@ -18,432 +19,2571 @@ msgstr ""
...
+msgstr ""
+
+#: 3.0/applet.js:167 3.0/applet.js:169 3.0/applet.js:171 3.0/applet.js:174
+#: 3.0/applet.js:177 3.0/applet.js:178 3.0/applet.js:179 3.0/applet.js:180
+#: 3.8/weather-applet.js:17878 3.8/weather-applet.js:17880
+#: 3.8/weather-applet.js:17882 3.8/weather-applet.js:17885
+#: 3.8/weather-applet.js:17888 3.8/weather-applet.js:17889
+#: 3.8/weather-applet.js:17890 3.8/weather-applet.js:17891
+msgid "Service Error"
+msgstr ""
```
[the remark this answers is not in the document]
Comment (rcalixte) [reply weather@mockturtl/files/weather@mockturtl/po/ca.po:34]: The output strings will match the source strings until they are properly translated. End users will see English, not empty strings, until there is localization support for their locale.

**Comment by Gr3q** [comment]
Thank you.

> And on Fedora 40, with Cinnamon 6.0.4, the location to install the app is apparently ~/.local/share/cinnamon/applets

Yes indeed, the configs go into `.config/cinnamon`. It doesn't seem to respect XDG base directory spec but that's irrelevant now.

> I was able to install the new version, and it appears to be refreshing correctly, though I had to manually add my location to make it work. The app didn't obtain my location automatically and gave the circular refreshing icon until I clicked the manual toggle.

It seems the GeoIP library was the one never returning, essentially locking the whole app if it can't get the location for some reason. Now it times out after 30 seconds. So I will look into it why, why doesn't it timeout on it's own and why the applet doesn't fall back to something that's working.

In the meantime, could you give me the output of this? `sudo journalctl -exau geoclue`

**Comment by Gr3q** [comment]
Apart from GeoCode and GeoClue are crazy slow, I also failed to handle GeoCode errors properly (try/catch outside the callback doesn't handle errors).

It seems the error is that Gnome's Nominatim (https://nominatim.gnome.org/docs) instance GeoCode is using by default is permanently fucked. I changed it to use https://nominatim.openstreetmap.org because that works properly.

**Review by rcalixte — APPROVED** [review]

**Pull request merged**